crates/astria-sequencer/src/app/mod.rs

```diff
@@ -175,6 +180,48 @@ impl From<abci::request::ProcessProposal> for ProposalFingerprint {
     }
 }
 
+#[derive(Clone)]
+pub(crate) struct OptimisticBlockChannels {
+    optimistic_block_sender: tokio::sync::watch::Sender<Option<SequencerBlock>>,
+    committed_block_sender: tokio::sync::watch::Sender<Option<SequencerBlockCommit>>,
+}
+
+impl OptimisticBlockChannels {
+    pub(crate) fn new(
+        optimistic_block_sender: tokio::sync::watch::Sender<Option<SequencerBlock>>,
+        committed_block_sender: tokio::sync::watch::Sender<Option<SequencerBlockCommit>>,
+    ) -> Self {
+        Self {
+            optimistic_block_sender,
+            committed_block_sender,
+        }
+    }
+
+    pub(crate) fn optimistic_block_sender(&self) -> Sender<Option<SequencerBlock>> {
+        self.optimistic_block_sender.clone()
+    }
+
+    pub(crate) fn committed_block_sender(&self) -> Sender<Option<SequencerBlockCommit>> {
+        self.committed_block_sender.clone()
+    }
+
+    pub(crate) fn send_optimistic_block(&self, block: Option<SequencerBlock>) {
+        if self.optimistic_block_sender().receiver_count() > 0 {
+            if let Err(e) = self.optimistic_block_sender.send(block) {
+                error!(error = %e, "failed to send optimistic block");
+            }
+        }
+    }
+
+    pub(crate) fn send_committed_block(&self, block: Option<SequencerBlockCommit>) {
+        if self.committed_block_sender().receiver_count() > 0 {
+            if let Err(e) = self.committed_block_sender.send(block) {
+                error!(error = %e, "failed to send committed block");
```

Comment:
we should not error out `process_proposal` or `finalize_block` if sending a block on the watch fails